Bottom: 4cc3c7ed4f972785c34916efc6d2bd245bb2cac6
Top:    d302fa38e05a4a1699d6594f0ece742f57ba8f57
Author: Ben Greear <greearb@candelatech.com>
Date:   2014-06-11 13:05:13 -0700

wireless: improve dfs-region intersection.

If one is UN-SET, use the other.  Seems this would
be more correct that what we have now.

Signed-off-by: Ben Greear <greearb@candelatech.com>


---

diff --git a/net/wireless/reg.c b/net/wireless/reg.c
index 5dbac37..cdf96ae 100644
--- a/net/wireless/reg.c
+++ b/net/wireless/reg.c
@@ -806,8 +806,15 @@ static enum nl80211_dfs_regions
 reg_intersect_dfs_region(const enum nl80211_dfs_regions dfs_region1,
 			 const enum nl80211_dfs_regions dfs_region2)
 {
-	if (dfs_region1 != dfs_region2)
+	if (dfs_region1 != dfs_region2) {
+		pr_info("intersect-dfs-region, region1: %d  region2: %d\n",
+			dfs_region1, dfs_region2);
+		if (dfs_region1 == NL80211_DFS_UNSET)
+			return dfs_region2;
+		if (dfs_region2 == NL80211_DFS_UNSET)
+			return dfs_region1;
 		return NL80211_DFS_UNSET;
+	}
 	return dfs_region1;
 }
